### ISSUE TITLE
Safari: User name input does not lead to listing of repos

### ISSUE THREAD

**Issue by michaelaye** [jupyterlab/jupyterlab-github#54]
Using J-Lab in Safari with this plugin, i put a username and it does not show the available repos.
I verified via Chrome, that the plugin was installed successfully, as things work there.

Here's what the javascript console says:

![screenshot 2018-04-26 11 27 11](https://user-images.githubusercontent.com/69774/39319343-6abcf956-4946-11e8-895a-499fc267dfd2.png)

Here's my jupyters:
```bash
(stable) └─❱❱❱ conda list jupyter                                       +7857 11:46 ❰─┘
# packages in environment at /Users/klay6683/miniconda3/envs/stable:
#
# Name                    Version                   Build  Channel
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                    py36_0    conda-forge
jupyter_console           5.2.0                    py36_0    conda-forge
jupyter_contrib_core      0.3.3                    py36_1    conda-forge
jupyter_contrib_nbextensions 0.5.0                    py36_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyter_highlight_selected_word 0.2.0                    py36_0    conda-forge
jupyter_latex_envs        1.4.4                    py36_0    conda-forge
jupyter_nbextensions_configurator 0.4.0                    py36_0    conda-forge
jupyterlab                0.32.0                   py36_1    conda-forge
jupyterlab_launcher       0.10.5                   py36_0    conda-forge
```
and my `conda info`:
```bash
(stable) └─❱❱❱ conda info                                               +7858 11:47 ❰─┘

     active environment : stable
    active env location : /Users/klay6683/miniconda3/envs/stable
            shell level : 1
       user config file : /Users/klay6683/.condarc
 populated config files : /Users/klay6683/.condarc
          conda version : 4.5.1
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : /Users/klay6683/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/menpo/osx-64
                          https://conda.anaconda.org/menpo/noarch
                          https://conda.anaconda.org/jlaura/osx-64
                          https://conda.anaconda.org/jlaura/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/conda-canary/osx-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://conda.anaconda.org/NSIDC/osx-64
                          https://conda.anaconda.org/NSIDC/noarch
          package cache : /Users/klay6683/miniconda3/pkgs
                          /Users/klay6683/.conda/pkgs
       envs directories : /Users/klay6683/miniconda3/envs
                          /Users/klay6683/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.1 requests/2.18.4 CPython/3.6.5 Darwin/16.7.0 OSX/10.12.6
                UID:GID : 273771:2260
             netrc file : None
           offline mode : False
```

**Comment by ian-r-rose** [comment]
Yikes, I am not sure what is happening there (I don't have a Mac, so it's tough for me to check). Does [this](https://support.apple.com/en-us/HT203370) do anything useful?

**Comment by michaelaye** [comment]
I deleted the cookies and cache for `localhost` but nothing changed.

**Comment by vidartf** [comment]
@michaelaye If you run `jupyter lab --debug`, does the server log include anything about the redirects?

**Comment by michaelaye** [comment]
Some 200 GET /api/ stuff but no errors, and it seems to be unrelated.

**Comment by michaelaye** [comment]
Note, that I have the same problem on 2 different OSX machines, so it's not a mere single-machine upset.

**Comment by ian-r-rose** [comment]
Does it work if you remove the server extension (`jupyterlab_github`)? You will likely get rate-limited after a few minutes, but that will allow us to narrow down what is happening.

**Comment by michaelaye** [comment]
Please advise on how to remove the server extension. I only ever saw install commands. ;)

**Comment by ian-r-rose** [comment]
Sure thing :)

You can temporarily disable it by running `jupyter serverextension disable jupyterlab_github`, and then re-enable it by running `jupyter serverextension enable jupyterlab_github`

Also, can I ask whether you have set this up using the (now deprecated) OAuth app approach, or the access token approach?

**Comment by michaelaye** [comment]
I have a "jupyterlab" entry in my OAuth apps in my GH Developer settings, I guess that means it's via OAuth, if the access point is called "jupyterlab" ?

**Comment by ian-r-rose** [comment]
More specifically, which entries do you have in your `jupyter_notebook_config.py` file, as described [here](https://github.com/jupyterlab/jupyterlab-github#3-installing-the-serverextension)?

**Comment by michaelaye** [comment]
i do have client_id and client_sceret defined in my notebook config, while I have the OAuth jupyterlab access defined in my GH developer settings. note that this setup works while using Chrome.

**Comment by michaelaye** [comment]
but of course i can remove the OAuth part in GH if you think it's a problem.

**Comment by ian-r-rose** [comment]
I don't know that it's a problem, but we recently changed the recommended approach to using an access token, rather than `client_id` and `client_secret`. I wonder if the same problem arises using that.

**Comment by michaelaye** [comment]
i disabled the serverextension:

```
(stable) └─( 2 ) ❱❱❱ jupyter serverextension disable jupyterlab_github                      +9664 14:03 ❰─┘
Disabling: jupyterlab_github
- Writing config: /Users/klay6683/.jupyter
```
and it still does not work in Safari, while it still works in Chrome.

**Comment by ian-r-rose** [comment]
When disabled, did the warnings on the javascript console change?

**Comment by michaelaye** [comment]
The promise rejection isn't there:

<img width="670" alt="screenshot 2018-05-10 14 08 35" src="https://user-images.githubusercontent.com/69774/39891772-bd1e3bc2-545b-11e8-89db-7e1d25d81709.png">

Note, that these lines were already there **BEFORE** I entered a user name, just with the Github tab open, is that to be expected?

**Comment by ian-r-rose** [comment]
*Interesting.* The redirects appear to be happening in the interaction with the notebook server, not with GitHub itself. That request is a test to see if the `jupyterlab_github` extension is present, and it is not even getting past that stage in Safari.

Can you try opening a new browser tab with that link in it to see what happens?

**Comment by ian-r-rose** [comment]
Okay, well, that's at least consistent. For a more minimal case, what happens if you remove the query parameter, and just go to `https://localhost:8901/github`?

**Comment by michaelaye** [comment]
Then I get a 404: Not found.

**Comment by ian-r-rose** [comment]
Okay, we may be making progress :). It seems like Safari may be treating the trailing slash differently than any other browsers. One more thing to check: how about
`https://localhost:8901/github?verylongandconfusingstring`?

**Comment by michaelaye** [comment]
404

**Comment by ian-r-rose** [comment]
I just opened a PR with a possible workaround, any chance you would be able to test if it fixes this? No rush if you can't find time.

**Comment by michaelaye** [comment]
no change here after installing your PR. Can you point out what slash are you talking about?

**Comment by ian-r-rose** [comment]
Hm, that's unfortunate. I am referring to the slash between the url and the query string. That is to say, I thought that Safari might be getting choked on
```
 https://localhost:8901/github/?verylongandconfusingstring
```
vs.
```
 https://localhost:8901/github?verylongandconfusingstring?
```

**Comment by michaelaye** [comment]
I still have a slash in that URL.
Here was my installation procedure (from the README.md of your fork):

1. jupyter labextension install @jupyterlab/github
2. Did nothing to my jupyter_config where I have client and secret already in
3. In your `remove_trailing_slash` I did `pip install -e .`

As I said, in the Javascript console, I still see:

https://localhost:8901/github/?1526166461199

**Comment by ian-r-rose** [comment]
Ah, it looks like you still installed the published version. If you have cloned the fork, can you try installing the local version by running
```
jlpm install
jlpm run build
jupyter labextension install .
```
from the repo directory?

**Comment by michaelaye** [comment]
oh, darn, you wrote that! Sorry, forgot...

**Comment by ian-r-rose** [comment]
Can you navigate to and open files? Some of those are expected (though I don't think that it's necessary for Safari to output them all).

**Comment by michaelaye** [comment]
yes, i can navigate to notebooks and run them.

**Comment by michaelaye** [comment]
Thank you for debugging this with me! So I guess we can close this? I'm guessing the unauthorized repos are from my other organizations?

**Comment by ian-r-rose** [comment]
Thanks for all the help in tracking it down! I can merge the fix and release a patch version today. I'm not entirely sure *why* Safari chokes on those URLs, but the fix is innocuous enough.

Those unauthorized errors are actually expected (though I wish Safari would not spam the console with them, maybe I can find a way around that). They are some checks for things at various endpoints, along with fallbacks if they fail.

**Comment by ian-r-rose** [comment]
Fixed by #56
